GSA-TTS/FAC
# 3571 handle accidental uploads of findings associated workbooks

## Description
This PR allows users to delete a worksheet after they have uploaded it and realized it is not needed.
The change only applies to optional worksheets. When users upload an optional worksheet, a new link appears near the edit link to allow users to delete the worksheet.
Upon clicking on the worksheet removal link, a new page is displayed, warning the user about the deletion action and asking for confirmation before proceeding. If confirmed, the worksheet will be deleted and the user will be redirected to the submission progress page. In case of an error, the application stays on the deletion page and displays a user-friendly error message to indicate the issue.
See #3571 for context.

## To Test:
- Upload an optional worksheet. Note that the removal link appears.
- Click on the link and delete the worksheet.
- Confirm the deletion when prompted.
- Verify that you are redirected to the submission progress page.
- Ensure that the worksheet is correctly removed.

Noticed new links from the submission checklist page 
![image](https://github.com/GSA-TTS/FAC/assets/26260777/35c8391e-2a0f-4b15-a28d-28d3cb79ef2a)

And from the upload page

![image](https://github.com/GSA-TTS/FAC/assets/26260777/ba764dc7-8111-4695-b545-b4c0f7ef63ce)

## PR checklist: submitters


## PR checklist: reviewers


The larger the PR, the stricter we should be about these points.
